### PR TITLE
here-docs code improvement for GHA summary to maven-cd.yml runs (#33)

### PR DESCRIPTION
**Pull request by mhoffrog**
@jglick - Following your comment on #33 

No functional change

## Test
s. https://github.com/mavenplugins/jenkins-unleash-plugin-fork/actions/runs/10962303075

tested after code improvements co-authored by @jglick :
https://github.com/mavenplugins/jenkins-unleash-plugin-fork/actions/runs/10965939050

### CODE REVIEW

File: .github/workflows/maven-cd.yml [hunk 1]
```diff
@@ -51,17 +51,22 @@ jobs:
...
-          echo "================================"                 | tee -a $GITHUB_STEP_SUMMARY
-          echo "Release job filter details:"                      | tee -a $GITHUB_STEP_SUMMARY
-          echo "          VALIDATE_ONLY: ${VALIDATE_ONLY}"        | tee -a $GITHUB_STEP_SUMMARY
-          echo "              CI_STATUS: ${CI_STATUS}"            | tee -a $GITHUB_STEP_SUMMARY
-          echo "   INTERESTING_CATEGORY: ${INTERESTING_CATEGORY}" | tee -a $GITHUB_STEP_SUMMARY
-          echo "--------------------------------"                 | tee -a $GITHUB_STEP_SUMMARY
-          echo "=>       SHOULD_RELEASE: ${SHOULD_RELEASE}"       | tee -a $GITHUB_STEP_SUMMARY
-          echo "================================"                 | tee -a $GITHUB_STEP_SUMMARY
-          echo "\`\`\`"                                                 >> $GITHUB_STEP_SUMMARY
+          MD_CODE="\`\`\`"
```
Comment (jglick) [inline .github/workflows/maven-cd.yml]:
```suggestion
          MD_CODE='```'
```

File: .github/workflows/maven-cd.yml [hunk 2]
```diff
@@ -51,17 +51,22 @@ jobs:
...
+                    VALIDATE_ONLY: ${VALIDATE_ONLY}
+                        CI_STATUS: ${CI_STATUS}
+             INTERESTING_CATEGORY: ${INTERESTING_CATEGORY}
+          --------------------------------
+          =>       SHOULD_RELEASE: ${SHOULD_RELEASE}
+          ================================
+          ${MD_CODE}
+          END_OF_SUMMARY
+          )
+          echo "${GHA_SUMMARY}" | tee -a $GITHUB_STEP_SUMMARY
```
Comment (jglick) [inline .github/workflows/maven-cd.yml]: More simply (untested),

```suggestion
          cat << END_OF_SUMMARY | tee -a $GITHUB_STEP_SUMMARY
          ## Job Filter Summary
          ${MD_CODE}
          ================================
          Release job filter details:
                    VALIDATE_ONLY: ${VALIDATE_ONLY}
                        CI_STATUS: ${CI_STATUS}
             INTERESTING_CATEGORY: ${INTERESTING_CATEGORY}
          --------------------------------
          =>       SHOULD_RELEASE: ${SHOULD_RELEASE}
          ================================
          ${MD_CODE}
          END_OF_SUMMARY
```

**Review by jglick — APPROVED** [review]
Tested?

File: .github/workflows/maven-cd.yml (hunk 2 above)
Comment (mhoffrog) [reply .github/workflows/maven-cd.yml]: `cat << END_OF_SUMMARY | tee -a $GITHUB_STEP_SUMMARY` - this is what I started with, but for some reason it did not get logged to stdout - only step summary was filled. Did test it now within a bash shell and there it worked. Will commit it but will perform another GHA test run before giving green light.

**Comment by mhoffrog** [comment]
@jglick Let me do another test run before merging.

**Comment by mhoffrog** [comment]
> Tested?

@jglick - it is tested and working as expected! Many thanks for your input!

For test result please look at: https://github.com/mavenplugins/jenkins-unleash-plugin-fork/actions/runs/10965939050

**Review by jglick — APPROVED** [review]

**Comment by mhoffrog** [comment]
@timja - FYI - should be released when there is some time.

**Comment by timja** [comment]
Done

**Pull request merged**